clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/client/P2PFileTransferClientHandler.java

```diff
@@ -66,9 +78,37 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Ex
       // Already end of transfer. Close the connection and completes the future
       if (response.headers().get(BLOB_TRANSFER_STATUS) != null
           && response.headers().get(BLOB_TRANSFER_STATUS).equals(BLOB_TRANSFER_COMPLETED)) {
+        LOGGER.info("Received blob transfer completed response.");
+        resetState();
         handleEndOfTransfer(ctx);
         return;
       }
+      currentTransferType = response.headers().get(BLOB_TRANSFER_TYPE);
```

Comment:
The logic here is: if the msg is a file or metadata, we need to update the current type so that we can decide if to pass it to the handlerMetadataTransferResponse or handleFileTransferResponse. It has to be in the `if` statement to check if it is httpResponse. It is not related to BLOB_TRANSFER_COMPLETED.